step-framework-parent/step-framework-timeseries/src/main/java/step/core/timeseries/TimeSeriesFilterBuilder.java

```diff
@@ -0,0 +1,68 @@
+package step.core.timeseries;
+
+import step.core.collections.Filter;
+import step.core.collections.Filters;
+import step.core.timeseries.aggregation.TimeSeriesAggregationQuery;
+import step.core.timeseries.query.OQLTimeSeriesFilterBuilder;
+import step.core.timeseries.query.TimeSeriesQuery;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static step.core.timeseries.TimeSeriesConstants.ATTRIBUTES_PREFIX;
+import static step.core.timeseries.TimeSeriesConstants.TIMESTAMP_ATTRIBUTE;
+
+public class TimeSeriesFilterBuilder {
+
+    public static Filter buildFilter(String oql) {
+        return OQLTimeSeriesFilterBuilder.getFilter(oql);
+    }
+
+    public static Filter buildFilter(Long from, Long to) {
+        ArrayList<Filter> filters = new ArrayList<>();
+        if (from != null) {
+            filters.add(Filters.gte(TIMESTAMP_ATTRIBUTE, from));
+        }
+        if (to != null) {
+            filters.add(Filters.lt(TIMESTAMP_ATTRIBUTE, to));
+        }
+        return Filters.and(filters);
+    }
+
+    public static Filter buildFilter(Map<String, Object> attributes) {
+        if (attributes == null || attributes.isEmpty()) {
+            return Filters.empty();
+        }
+        List<Filter> filters = attributes.entrySet().stream()
+                .map(e -> {
+                    Object value = e.getValue();
+                    if (value instanceof Boolean) {
+                        return Filters.equals(ATTRIBUTES_PREFIX + e.getKey(), (Boolean) e.getValue());
```

Comment:
- Use value instead of recalling e.getValue()
- Declare variable key for e.getKey()